### PR TITLE
Add GetProperties to navigation and slam API

### PR DESCRIPTION
**Pull request by jeremyrhyde**
This PR adds `GetProperties` to the SLAM and navigation services. A future PR will handle the removal of `GetLatestMapInfo`

**Updated Scope Doc:**  https://docs.google.com/document/d/1TRkg9pqNoDXT-c52uXXV9g1-IR80HuI1eqRQIu-AW9k/edit

**JIRA Ticket:** https://viam.atlassian.net/browse/RSDK-6031

### CODE REVIEW

File: proto/viam/service/navigation/v1/navigation.proto [hunk 1]
```diff
@@ -59,6 +59,12 @@ service NavigationService {
     };
   }
 
+  rpc GetProperties(GetPropertiesRequest) returns (GetPropertiesResponse) {
```
Comment (randhid) [inline proto/viam/service/navigation/v1/navigation.proto:62]: Add a comment over this for the API generated docs. Just a simple explanation of what it is.

Don't do this for the other rpcs, we'll add incrementally and i don't want to write anything wrong.

File: proto/viam/service/slam/v1/slam.proto [hunk 2]
```diff
@@ -45,6 +45,13 @@ service SLAMService {
     };
   }
 
+  // GetProperties returns properties, such as mapping_mode, of the current slam service
```
Comment (randhid) [inline proto/viam/service/slam/v1/slam.proto:48]: Add comments for the mapping mode types here and above the enum! Well done adding a comment to the slam service :D

File: proto/viam/service/slam/v1/slam.proto [hunk 3]
```diff
@@ -112,3 +126,14 @@ message GetLatestMapInfoRequest {
 message GetLatestMapInfoResponse {
```
Comment (randhid) [inline proto/viam/service/slam/v1/slam.proto:126]: .....

**Review by randhid — APPROVED** [review]
Few nits! But Let's go!

File: proto/viam/service/navigation/v1/navigation.proto (hunk 1 above)
Comment (jeremyrhyde) [reply proto/viam/service/navigation/v1/navigation.proto:64]: Done!

**Review by JohnN193 — APPROVED** [review]

**Pull request merged**